sdk/communication/azure-communication-callautomation/src/main/java/com/azure/communication/callautomation/implementation/converters/CommunicationIdentifierConverter.java

```diff
@@ -33,6 +34,39 @@ public static CommunicationIdentifier convert(CommunicationIdentifierModel ident
         assertSingleType(identifier);
         String rawId = identifier.getRawId();
 
+        CommunicationIdentifierModelKind kind = identifier.getKind();
+
+        if (kind != null) {
+            if (kind == CommunicationIdentifierModelKind.COMMUNICATION_USER
+                && identifier.getCommunicationUser() != null) {
+                Objects.requireNonNull(identifier.getCommunicationUser().getId());
+                return new CommunicationUserIdentifier(identifier.getCommunicationUser().getId());
+            }
+
+            if (kind == CommunicationIdentifierModelKind.PHONE_NUMBER
+                && identifier.getPhoneNumber() != null) {
+                PhoneNumberIdentifierModel phoneNumberModel = identifier.getPhoneNumber();
+                Objects.requireNonNull(phoneNumberModel.getValue());
+                return new PhoneNumberIdentifier(phoneNumberModel.getValue()).setRawId(rawId);
+            }
+
+            if (kind == CommunicationIdentifierModelKind.MICROSOFT_TEAMS_USER
+                && identifier.getMicrosoftTeamsUser() != null) {
+                MicrosoftTeamsUserIdentifierModel teamsUserIdentifierModel = identifier.getMicrosoftTeamsUser();
+                Objects.requireNonNull(teamsUserIdentifierModel.getUserId());
+                Objects.requireNonNull(teamsUserIdentifierModel.getCloud());
+                Objects.requireNonNull(rawId);
+                return new MicrosoftTeamsUserIdentifier(teamsUserIdentifierModel.getUserId(),
+                    teamsUserIdentifierModel.isAnonymous())
+                    .setRawId(rawId)
```

Comment:
in the constructor of the MicrosoftTeamsUserIdentifier the rawId is set, so I guess there is no need to reset it again

Reply:
As far as I understand, only userId is set in constructor:
`    public MicrosoftTeamsUserIdentifier(String userId, boolean isAnonymous) {
        this.rawIdSet = false;
        this.cloudEnvironment = CommunicationCloudEnvironment.PUBLIC;
        if (CoreUtils.isNullOrEmpty(userId)) {
            throw new IllegalArgumentException("The initialization parameter [userId] cannot be null or empty.");
        } else {
            this.userId = userId;
            this.isAnonymous = isAnonymous;
            this.generateRawId();
        }
    }`

`setRawId` method is called from a super class and sets a raw ID
    `    public MicrosoftTeamsUserIdentifier setRawId(String rawId) {
        super.setRawId(rawId);
        this.rawIdSet = true;
        return this;
    }`

Reply:
I think the value would be the same as this generating logic is approved solution and covered by tests. @petrsvihlik wdyt about setting it again?

Reply:
Currently raw ID is generated based on user ID and cloud environment only. Under some circumstances, rawId from `CommunicationIdentifierModel` could be a totally different value (like `123abc`), which could not be generated. I'm not sure what expected behaviour is but not setting a rawID explicitly here will change the previous logic.

Reply:
This is an exception for PhoneNumbers and MSTeamsUsers. The rawId must be set explicitly to support some legacy identifiers.